src/actions/AddTallyAction.C

```diff
@@ -89,13 +99,15 @@ AddTallyAction::addMeshTally(unsigned int instance, const Point & translation)
   std::string obj_name = _name;
   if (_mesh_translations.size() > 1)
   {
-    obj_name += "_" + Moose::stringify(instance);
+    if (instance != 0)
```

Comment:
IMO we should still print the zero instance if there is > 1 instance (but I agree, it can be suppressed if there's only ever one instance)